scripts/db_migrator.py

```diff
@@ -567,6 +567,25 @@ def migrate_telemetry(self):
         if not certs:
             self.configDB.set_entry("TELEMETRY", "certs", telemetry_data.get("certs"))
 
+    def migrate_gnmi(self):
+        # GNMI - add missing key
+        if not self.minigraph_data or 'GNMI' not in self.minigraph_data:
+            # If there's no GNMI in minigraph, copy configuration from CONFIG_DB TELEMETRY table
```

Comment:
>If there's no GNMI in minigraph, copy configuration from CONFIG_DB TELEMETRY table [](http://example.com/codeflow?start=14&length=82)

Could you double check if this works with L2 switch mode? [https://github.com/sonic-net/SONiC/wiki/L2-Switch-mode](https://github.com/sonic-net/SONiC/wiki/L2-Switch-mode)

Reply:
We previously has a bug that mistakenly generating config for telemetry in L2 mode.